tests/bot/cogs/moderation/test_utils.py

```diff
@@ -0,0 +1,411 @@
+                "post_result": "bar",
+                "raise_error": None,
+                "payload": {
+                    "discriminator": some_mock.discriminator,
+                    "id": some_mock.id,
+                    "in_guild": False,
+                    "name": some_mock.name,
+                    "roles": []
+                }
+            }
+        ]
+
+        for case in test_cases:
+            test_user = case["user"]
+            expected = case["post_result"]
+            error = case["raise_error"]
+            payload = case["payload"]
+
+            with self.subTest(user=test_user, result=expected, error=error, payload=payload):
+                log_mock.reset_mock()
+                self.bot.api_client.post.reset_mock(side_effect=True)
+                self.ctx.bot.api_client.post.return_value = expected
+
+                self.ctx.bot.api_client.post.side_effect = error
+
+                result = await utils.post_user(self.ctx, test_user)
+
+                if error:
+                    self.assertIsNone(result)
+                else:
+                    self.assertEqual(result, expected)
+
+                if not error:
+                    self.bot.api_client.post.assert_awaited_once_with("bot/users", json=payload)
+                else:
+                    self.ctx.send.assert_awaited_once()
+                    self.assertTrue(str(error.status) in self.ctx.send.call_args[0][0])
+
+                if isinstance(test_user, MagicMock):
+                    log_mock.debug.assert_called_once()
+                else:
+                    log_mock.debug.assert_not_called()
+
+    async def test_send_private_embed(self):
+        """Should DM the user and return `True` on success or `False` on failure."""
+        embed = Embed(title="Test", description="Test val")
+
+        test_cases = [
+            {
+                "expected_output": True,
+                "raised_exception": None
+            },
+            {
+                "expected_output": False,
+                "raised_exception": HTTPException(AsyncMock(), AsyncMock())
+            },
+            {
+                "expected_output": False,
+                "raised_exception": Forbidden(AsyncMock(), AsyncMock())
+            },
+            {
+                "expected_output": False,
+                "raised_exception": NotFound(AsyncMock(), AsyncMock())
+            }
+        ]
+
+        for case in test_cases:
+            expected = case["expected_output"]
+            raised = case["raised_exception"]
+
+            with self.subTest(expected=expected, raised=raised):
+                self.user.send.reset_mock(side_effect=True)
+                self.user.send.side_effect = raised
+
+                result = await utils.send_private_embed(self.user, embed)
+
+                self.assertEqual(result, expected)
+                if expected:
+                    self.user.send.assert_awaited_once_with(embed=embed)
```

Comment:
Why only check `if expected`? Isn't `send` being called unconditionally?

Reply:
Mock is awaited only when no errors raised.

Reply:
Is that so? Try removing the conditional - test passes just fine.

Correct me if I'm wrong, but with `self.user.send` being an `AsyncMock,` the side effect isn't actioned until after the `send` coro is awaited.

Reply:
For me, the test doesn't pass when I remove conditional: this says that it's awaited 0 times.